common/viewer.cpp

```diff
@@ -442,30 +442,35 @@ namespace rs2
                  s.second.profile.stream_type() == RS2_STREAM_DEPTH ||
                  s.second.profile.stream_type() == RS2_STREAM_FISHEYE))
             {
+                auto profile_unique_id = s.second.profile.unique_id();
+                bool profile_found = streams_origin.find(profile_unique_id) != streams_origin.end();
+
                 if (selected_tex_source_uid == -1 && selected_depth_source_uid != -1)
                 {
-                    if (streams_origin.find(s.second.profile.unique_id()) != streams_origin.end() &&
-                        streams.find(streams_origin[s.second.profile.unique_id()]) != streams.end())
+                    if (profile_found && streams.find(streams_origin[profile_unique_id]) != streams.end())
                     {
-                        selected_tex_source_uid = streams_origin[s.second.profile.unique_id()];
+                        selected_tex_source_uid = streams_origin[profile_unique_id];
                     }                         
                 }
-                if ((streams_origin.find(s.second.profile.unique_id()) != streams_origin.end() &&streams_origin[s.second.profile.unique_id()] == selected_tex_source_uid))
+                if ((profile_found && streams_origin[profile_unique_id] == selected_tex_source_uid))
                 {
                     selected_tex_source = i;
                 }
 
-                // The texture source shall always refer to the raw (original) streams
-                tex_sources.push_back(streams_origin[s.second.profile.unique_id()]);
-                tex_profiles.push_back(s.second.profile);
+                if ( profile_found )
```

Comment:
Any danger here? What does it mean, if the profile wasn't found?

Reply:
This function is called in a loop, if the profile is not found it will not be display to the user as an option, once it will be there it will be displayed.